sklearn/utils/_print_versions.py

```diff
@@ -0,0 +1,118 @@
+"""
+Utility methods to print system info for debugging
+
+adapted from :func:`pandas.show_versions`
+"""
+# License: BSD 3 clause
+
+import platform
+import sys
+import importlib
+
+from .._build_utils import get_blas_info
```

Comment:
If you put this import inside the function that needs it, are you still getting the `TypeError`s ?

Reply:
\0/

now you have to explain to me why.. :)

```
$ ipython
Python 3.6.5 (default, Mar 30 2018, 06:42:10)
Type 'copyright', 'credits' or 'license' for more information
IPython 6.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import sklearn

In [2]: sklearn.show_versions()
ATLAS version 3.10.1 built by aboucaud on Mar 21 jan 2014 15:28:46 CET:
   UNAME    : Darwin glx-boucaud.ias.u-psud.fr 13.0.2 Darwin Kernel Version 13.0.2: Sun Sep 29 19:38:57 PDT 2013; root:xnu-2422.75.4 1/RELEASE_X86_64 x86_64
   INSTFLG  : -1 0 -a 1 -l 1
   ARCHDEFS : -DATL_OS_OSX -DATL_ARCH_UNKNOWNx86 -DATL_CPUMHZ=2600 -DATL_SSE3 -DATL_SSE2 -DATL_SSE1 -DATL_USE64BITS -DATL_GAS_x8664
   F2CDEFS  : -DAdd_ -DF77_INTEGER=int -DStringSunStyle
   CACHEEDGE: 1572864
   F77      : /usr/local/bin/x86_64-apple-darwin13.0.2-gfortran-4.8, version GNU Fortran (GCC) 4.8.2
   F77FLAGS : -O -msse3 -m64
   SMC      : /usr/local/bin/gcc-4.7, version gcc-4.7 (GCC) 4.7.3
   SMCFLAGS : -O -fomit-frame-pointer -msse3 -m64
   SKC      : /usr/local/bin/gcc-4.7, version gcc-4.7 (GCC) 4.7.3
   SKCFLAGS : -O -fomit-frame-pointer -msse3 -m64

System
------
    python: 3.6.5 (default, Mar 30 2018, 06:42:10)  [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
executable: /Users/aboucaud/codes/python_venvs/sklearn-dev/bin/python3.6
   machine: Darwin-16.7.0-x86_64-i386-64bit

BLAS
----
    macros: HAVE_CBLAS=None, ATLAS_INFO="\"3.10.1\""
  lib_dirs: /usr/local/lib
cblas_libs: ptf77blas, ptcblas, atlas, ptf77blas, ptcblas

Python deps
-----------
       pip: 10.0.1
setuptools: 40.0.0
   sklearn: 0.20.dev0
     numpy: 1.14.5
     scipy: 1.1.0
    Cython: 0.28.4
    pandas: 0.23.3
matplotlib: None
```